src/actors/url-state.js

```diff
@@ -0,0 +1,61 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// @flow
```

Comment:
This is where some magic happens.

I wanted to a way to dispatch an action whenever the URL State changes (and so whenever the URL changes) to invalidate the current shortened URL.

Possible solutions without using an actor are:
* Doing it in `URLManager.js` would trigger a dreadful double render that I wanted to avoid.
* In `reducers/short-url.js`, react to the same actions as we do in `reducers/url-state.js`. But this is IMO error-prone as it's very easy to forget when adding new actions in the future.

In my opinion they are both bad. So my solution here is to subscribe to store changes. (see http://jamesknelson.com/join-the-dark-side-of-the-flux-responding-to-actions-with-actors/ for more background). This allows to dispatch a new action as a result of a state change. This is very powerful and can lead to infinite loop if not careful.

I also moved all of URLManager code here, because I need the URL to be changed before dispatching the action `URL_STATE_HAS_CHANGED`, because I need to be able to access it in `reducers/short-url.js`.

Reply:
My opinion here is that I don't think this problem requires adding another abstraction to the project. Actors sound like they would be a potential source of bugs, and make the update cycle more confusing. I'd prefer to wrap up this logic in the lifecycles of the component. This component has the benefit of being at the root of our components, so double updates won't really have performance problems as long as we check for `shouldComponentUpdate` and return `false`. None of the other connected components will re-render.

Barring all of that I would prefer to `getState` from a `ThunkAction` creator than add actors. I know that this is also not preferred, but we already are using them in the project.